### ISSUE TITLE
Parsing error: Unexpected token, expected ";" while trying to lint JSON files

### ISSUE THREAD

**Issue by slim-hmidi** [eslint/eslint#11593]
**Tell us about your environment**

* **ESLint Version:** 5.16.0
* **Node Version:** 10.13.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  env: {
    browser: true,
    es6: true,
  },
  extends: 'airbnb-base',
  globals: {
    Atomics: 'readonly',
    SharedArrayBuffer: 'readonly',
  },
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: 'module',
  },
  parser: "babel-eslint"
  
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
{
  "swagger": "2.0",
  "info": {
    "description": "This is a simple ecommerce application which contains some Restful APIs",
    "version": "1.0.0",
  },
  .
  .
  .
}
```

```bash
eslint ./server/* --ext .js,.json
```

**What did you expect to happen?**
the lint returns no error

**What actually happened? Please include the actual, raw output from ESLint.**
it returns an error:
```
2:12  error  Parsing error: Unexpected token, expected ";"

  1 | {
> 2 |   "swagger": "2.0",
    |            ^
  3 |   "info": {
  4 |     "description": "This is a simple ecommerce application which contains some Restful APIs",
  5 |     "version": "1.0.0",
```

**Are you willing to submit a pull request to fix this bug?**

**Comment by g-plane** [comment]
Hi @shim-hmidi , thanks for this issue.

It seems that you're trying linting JSON files, but I think babel-eslint doesn't support that.

**Comment by slim-hmidi** [comment]
Before I added it I used the default parser and I got this error.

**Comment by platinumazure** [comment]
@slim-hmidi The default parser only supports JavaScript (not JSON) as well. I think this is working as designed, although I understand your frustrations here.

I think you would need to look for a parser that supports JSON and converts it to ESTree for ESLint to consume. I don't know of one off the top of my head.

**Comment by slim-hmidi** [comment]
I found this parser [eslint-plugin-json](https://github.com/azeemba/eslint-plugin-json) but I don't know if it the suitable one.

**Comment by slim-hmidi** [comment]
@platinumazure the `eslint-plugin-json` fixs my problem. But I have a question:
If the default parser parses only the js files and we could lint a json extension with `--ext` option, we should modify the default parser to take a consideration the json files or add a plugin to parse them in the eslint configuration.
I don't find it normal to add json extension when lint files then I discover the default parser does not do the job. Maybe I said stupid things but I think by default it should parse all the allowed extensions.

**Comment by platinumazure** [comment]
@slim-hmidi ESLint allows any file extension to be specified. While ESLint's core parser and linting rules are designed for JavaScript, we intentionally designed ESLint to be pluggable. Parsers and plugins can be used to extend ESLint's functionality beyond pure JavaScript files.

I can understand how it might seem unintuitive when you're starting out and trying different CLI options; but the behavior you've observed is in line with the design and architecture of ESLint.
